### ISSUE TITLE
Module streams, missing cut() proc

### ISSUE THREAD

**Issue by StefanSalewski** [nim-lang/Nim#5559]
Would be really great to have a reliable one...

```
https://irclogs.nim-lang.org/02-03-2017.html
Short streams module question:
14:03:51	Salewski	import streams
14:03:57	Salewski	var s = newStringStream()
14:04:08	Salewski	s.write("aa bb cc dd"); s.setPosition(3) ;s.write("xx")
14:04:12	Salewski	echo s.data; s.close
14:04:20	Salewski	# output is "aa xx cc dd" # Is it possible to get "aa xx", something like s.setLen() ?
14:04:27	Salewski	done.
14:05:14	flyx	Salewski: well you can do setLen on s.data, but that would invalidate the pos
14:06:57	flyx	Salewski: write a proc cut(s: StringStream, at: int) = s.setPosition(at); s.data.setLen(at)
```

```
# hope that will work reliable
proc cut(s: var StringStream; pos: int) =
  assert(pos >= 0 and pos <= s.getPosition)
  s.setPosition(pos)
  s.data.setLen(pos)
  s.setPosition(pos)
```

**Comment by Araq** [comment]
What's the point? how does 'cut' work for an ordinary file stream?

**Comment by StefanSalewski** [comment]
I don't know. But I think when setPosition() is available, something like cut() should be possible also. At least I needed a cut operation for a string stream recently. But indeed maybe using just a plain string would have been the better solution. I used string stream, because I saw some people using it, one of these was Jason M. for his smart-gi work. And I think that I should better not admit that I had some trouble understanding the streams manual at all -- for example when flush() is necessary, or if first closing the stream and then writing string to file, or first writing to file and closing finally.
